### PR TITLE
Include fee in signed order

### PR DESCRIPTION
**Pull request by anxolin**
Depends on #52 

This PR takes a simple approach for fee calculation and include it in the signed order, it's meant to integrate with @W3stside PRs once we are 100% sure what's the best option

For now I think this will enable us to include the fee already in the posted order.

Includes:
- API call. Note I had to commented out because the API is not yet implemented
- Includes a utility to calculate the feeAmount depending on the relevant data
- include some unit tests for fee calculation
- Gets the fee before doing the signing, then includes it

### CODE REVIEW

File: src/custom/hooks/useSwapCallback.ts [hunk 1]
```diff
@@ -55,19 +56,31 @@ async function _postOrder(params: PostOrderParams): Promise<string> {
...
-  const selToken = path[0]
+  const sellToken = path[0]
   const buyToken = path[path.length - 1]
+  const sellAmount = inputAmount.raw.toString(10)
+  const buyAmount = outputAmount.raw.toString(10)
+
+  // TODO: This might disappear, and just take the state from the state after the fees PRs are merged
+  //  we assume, the solvers will try to satisfy the price, and this fee is just a minimal fee.
+  // Get Fee
+  const { feeRatio, minimalFee } = await getFeeInformation(chainId, sellToken.address)
```
Comment (Velenir) [inline src/custom/hooks/useSwapCallback.ts:67]: Yes, this should already be prefilled by fee middleware

File: src/custom/utils/operator.ts [hunk 2]
```diff
@@ -114,7 +115,8 @@ export async function postSignedOrder(params: { chainId: ChainId; order: OrderCr
   // Call API
   const baseUrl = _getApiBaseUrl(chainId)
   const response = await fetch(`${baseUrl}/orders`, {
-    ...POST_HEADERS,
+    ...DEFAULT_HEADERS,
```
Comment (Velenir) [inline src/custom/utils/operator.ts:118]: Either you use `DEFAULT_HEADERS` as `headers`,then do `headers: DEFAULT_HEADERS`, and change DEFAULTHEADERS shape
or call them `DEFAULT_OPTIONS` and then spread

File: src/custom/hooks/useSwapCallback.ts (hunk 1 above)
Comment (Velenir) [reply src/custom/hooks/useSwapCallback.ts:67]: If I understand the order correctly. Fee is added onto the order before posting but after user already sees the numbers without the fee.
That's not good

**Review by Velenir — APPROVED** [review]

File: src/custom/hooks/useSwapCallback.ts (hunk 1 above)
Comment (anxolin) [reply src/custom/hooks/useSwapCallback.ts:67]: This is an intermediate PR for the DEMO

I'm not 100% sure we have the calculations are right for the other PR. This PRs just walk in the direction of adding the fee. When the other PR is ready, we'll be able to put the two together

In the DEMO they will probably return 0 anyways

Comment (W3stside) [reply src/custom/hooks/useSwapCallback.ts:67]: i think this is a placeholder, otherwise agree

File: src/custom/utils/fee.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { BigNumber } from 'ethers'
+import { FeeInformation } from './operator'
+
+const BPS_BASE = 10_000 // The number of basis points to make up 100%.
+
+export interface GetFeeAmount extends Omit<FeeInformation, 'expirationDate'> {
+  amount: string
```
Comment (W3stside) [inline src/custom/utils/fee.ts:7]: can u change this to sellAmount or sth more clear?

**Review by W3stside — APPROVED** [review]
I'm approving but have concerns regarding how we are lying to users about fee (applying on post)

File: src/custom/utils/operator.ts (hunk 2 above)
Comment (anxolin) [reply src/custom/utils/operator.ts]: done!

**Comment by anxolin** [comment]
> I'm approving but have concerns regarding how we are lying to users about fee (applying on post)


- But this is only while we don't merge the other PR, this PRs goes in the right direction, right?
- He will see the fee in the signing modal: I agree is not nice for a "final product", but this is not the final PR
- The fee will be 0 anyways, so no lying to clients

**Pull request merged**